ibm/service/power/resource_ibm_pi_instance_action.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"log"
+	"time"
+)
+
+func ResourceIBMPIInstanceAction() *schema.Resource {
+	return &schema.Resource{
+
+		CreateContext: resourceIBMPIInstanceActionCreate,
+		ReadContext:   resourceIBMPIInstanceActionRead,
+		UpdateContext: resourceIBMPIInstanceActionUpdate,
+		DeleteContext: resourceIBMPIInstanceActionDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Update: schema.DefaultTimeout(15 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PI Cloud instance id",
+			},
+			Arg_PVMInstanceId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PVM instance ID",
+			},
+			Arg_PVMInstanceActionType: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"start", "stop", "hard-reboot", "soft-reboot", "immediate-shutdown", "reset-state"}),
+				Description:  "PVM instance action type",
+			},
+			Arg_PVMInstanceHealthStatus: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{PVMInstanceHealthOk, PVMInstanceHealthWarning}),
+				Default:      PVMInstanceHealthOk,
+				Description:  "Set the health status of the PVM instance to connect it faster",
+			},
+
+			// Computed
+			Attr_Status: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the PVM instance",
+			},
+			Attr_Progress: {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "The progress of an operation",
+			},
+			Attr_HealthStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The PVM's health status value",
+			},
+		},
+	}
+}
+
+func resourceIBMPIInstanceActionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	adiag := takeInstanceAction(ctx, d, meta, d.Timeout(schema.TimeoutCreate))
+	if adiag != nil {
+		return adiag
+	}
+
+	return resourceIBMPIInstanceActionRead(ctx, d, meta)
+}
+
+func resourceIBMPIInstanceActionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID, id, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	client := st.NewIBMPIInstanceClient(context.Background(), sess, cloudInstanceID)
+	powervmdata, err := client.Get(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
```

Comment:
Can we set other arguments also which are defined in schema like Arg_CloudInstanceID , Arg_PVMInstanceId...soon

Reply:
This is handled in the client, it will generate the get error message in `err` with the details.